site/src/pages/release-notes/1.23.0.mdx

```diff
@@ -0,0 +1,268 @@
+  error handler. #3421 #4716
+  ```java
+  ServiceErrorHandler handler = (ctx, cause) -> {
+    if (cause instanceof IllegalArgumentException) {
+      return HttpResponse.of(HttpStatus.BAD_REQUEST);
+    }
+    // Return null to let the default error handler handle the exception.
+    return null;
+  };
+
+  ServerBuilder sb = ...
+  sb.route()
+    .get("/foo")
+    .errorHandler(handler)
+    ...
+    .build();
+  ```
+- You can now handle an HTTP/1 request with an absolute URI
+  using <type://ServerBuilder#absoluteUriTransformer(Function)>. #4681 #4726
+  ```java
+  ServerBuilder sb = ...
+  sb.absoluteUriTransformer(absoluteUri -> {
+    // https://foo.com/bar -> /bar
+    return absoluteUri.replaceFirst("^https://\\.foo\\.com/", "/");
+    // Or store the original URI in a query and use it in the proxy service.
+    // return "/proxy?uri=" + URLEncoder.encode(absoluteUri);
+  });
+  ```
+- You can now listen to the result of DNS queries using <type://DnsQueryListener>. #4690 #4715
+  ```java
+  DnsQueryListener listener = new DnsQueryListener() {
+    @Override
+    void onSuccess(...) {...}
+    @Override
+    public void onFailure(...) {
+        logger.warn("DNS query failed");
+    }
+  }
+
+  DnsAddressEndpointGroup
+    .builder("foo.com")
+    .addDnsQueryListeners(listener)
+    .build();
+  ```
+- You can now create a <type://BlockingTaskExecutor> by wrapping a `ScheduledExecutorService`. #4760
+  - Additionally, you can create context-aware or context-propagating `Executor` and
+    <type://BlockingTaskExecutor>.
+  ```java
+  ScheduledExecutorService executorService = ...
+  // Wrap a ScheduledExecutorService.
+  BlockingTaskExecutor blockingTaskExecutor =
+    BlockingTaskExecutor.of(executorService);
+
+  RequestContext ctx = ...
+  BlockingTaskExecutor blockingTaskExecutor = ...
+  // context-aware
+  ContextAwareBlockingTaskExecutor.of(context, executor);
+  // context-propagating
+  RequestContext.makeContextPropagating(blockingTaskExecutor);
+  ```
+- You can now use <type://StreamMessage#of(InputStream)> to convert an `InputStream`
+  to <type://StreamMessage>. #3937 #4703
+  ```java
+  InputStream in = ...
+  // The input stream is divided by 8192 by default.
+  StreamMessage<HttpData> stream = StreamMessage.of(in);
+  ```
+- You can now use <type://StreamMessage#streaming()> to create <type://StreamWriter> for writing
+  streaming messages. #4253 #4696
+  ```java
+  StreamWriter<String> writer = StreamMessage.streaming();
+  writer.write("foo");
+  writer.write("bar");
+  writer.close();
+
+  // Subscribe to the writer.
+  writer.subscribe(...);
+  ```
+- You can now generate a <type://RequestId> using the information of <type://RoutingContext>. #4362 #4691
+  - Additionally, the request ID generator can be configured per-service or per-virtual host level. #4730 #4752
+  ```java
+  ServerBuilder sb = ...
+  sb.requestIdGenerator((routingCtx, req) -> {
+    return RequestId.of(routingCtx.headers().getLong("requestId"));
```

Comment:
How about using some real-world header such as the one from OpenTelemetry, Jaeger, etc?